internal/services/houseKeeper.go

```diff
@@ -0,0 +1,62 @@
+package services
+
+import (
+	"encoding/json"
+	"os"
+	"strconv"
+
+	k "github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/labstack/gommon/log"
+
+	"github.com/redhatinsights/ros-ocp-backend/internal/config"
+	"github.com/redhatinsights/ros-ocp-backend/internal/kafka"
+	"github.com/redhatinsights/ros-ocp-backend/internal/logging"
+	"github.com/redhatinsights/ros-ocp-backend/internal/model"
+	"github.com/redhatinsights/ros-ocp-backend/internal/types"
+	"github.com/redhatinsights/ros-ocp-backend/internal/utils/sources"
+)
+
+var cost_app_id int
```

Comment:
Would it be possible to use this value a function param instead of a global? I don't see much usage apart from sourcesListener.

Reply:
We cannot directly pass `cost_app_id` to the `sourcesListener`. If you see `StartHouseKeeperService()`function at the end it calls `kafka.StartConsumer()` and not `sourcesListener`. 
`kafka.StartConsumer()` has this signature - `StartConsumer(kafka_topic string, handler func(msg *kafka.Message))`. handler which in our case is `sourcesListener` should be a function which only has one argument of type - `*k.Message`

Reply:
`sourcesListener` is called every time when there is new event on kafka topic. If we call `GetCostApplicationID` inside the `sourcesListener` Then we will every time make an unnecessary API call to sources to get the cost_app_id. That is the reason I called `GetCostApplicationID` at the beginning when the service starts. 

Cost team also does the same - https://github.com/project-koku/koku/blob/main/koku/sources/kafka_listener.py#L388

Reply:
Ah I see, makes sense then